### PR TITLE
Same twoWayKey

### PR DESCRIPTION
**Pull request by fogelito**

### CODE REVIEW

File: tests/Database/Base.php [hunk 1]
```diff
@@ -3429,6 +3429,47 @@ public function testNoChangeUpdateDocumentWithoutPermission(Document $document):
         return $document;
     }
 
+    public function testRelationSameKey(): void
+    {
+    if (!static::getDatabase()->getAdapter()->getSupportForRelationships()) {
```
Comment (abnegate) [inline tests/Database/Base.php:3434]:
```suggestion
        if (!static::getDatabase()->getAdapter()->getSupportForRelationships()) {
```

File: tests/Database/Base.php [hunk 2]
```diff
@@ -3429,6 +3429,47 @@ public function testNoChangeUpdateDocumentWithoutPermission(Document $document):
...
+            return;
+        }
+
+        $permissions = [
+            Permission::read(Role::any()),
+            Permission::create(Role::any()),
+            Permission::delete(Role::any()),
+        ];
+
+        static::getDatabase()->createCollection("c1", [], [], $permissions);
```
Comment (abnegate) [inline tests/Database/Base.php:3445]: Let's use single quotes throughout the test

File: src/Database/Database.php [hunk 3]
```diff
@@ -1528,7 +1528,7 @@ public function createRelationship(
 
         $id ??= $relatedCollection->getId();
 
-        $twoWayKey ??= $collection->getId();
+        $twoWayKey ??= $collection->getId() . '_' . uniqid();
```
Comment (abnegate) [inline src/Database/Database.php:1531]: Maybe we could do this only after we get a duplicate exception? Then if there's no duplicate we still get a nice key

File: tests/Database/Base.php (hunk 2 above)
Comment (fogelito) [reply tests/Database/Base.php]: Done!

File: src/Database/Database.php [hunk 4]
```diff
@@ -1537,11 +1537,15 @@ public function createRelationship(
...
+            try {
+                if ($attribute->getAttribute('type') === self::VAR_RELATIONSHIP
+                    && \strtolower($attribute->getAttribute('options')['twoWayKey']) === \strtolower($twoWayKey)
+                    && $attribute->getAttribute('options')['relatedCollection'] === $relatedCollection->getId()
+                ) {
+                    throw new DuplicateException('Related attribute already exists');
+                }
+            } catch (DuplicateException $e) {
+                $twoWayKey ??= $collection->getId() . '_' . uniqid();
             }
```
Comment (abnegate) [inline src/Database/Database.php:1549]: I don't think we need a try/catch here, we can use use the if:

```suggestion
            if ($attribute->getAttribute('type') === self::VAR_RELATIONSHIP
                && \strtolower($attribute->getAttribute('options')['twoWayKey']) === \strtolower($twoWayKey)
                && $attribute->getAttribute('options')['relatedCollection'] === $relatedCollection->getId()
            ) {
                $twoWayKey ??= $collection->getId() . '_' . uniqid();
            }
```

**Comment by fogelito** [comment]
We tool care of this on Appwrite side

**Pull request closed without merging**